pkg/diagnose/deployment_metrics.go

```diff
@@ -0,0 +1,88 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnose
+
+import (
+	"strings"
+
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/pkg/cluster"
+)
+
+const (
+	gwCurlMetricsCommand = "curl -I -m 10 submariner-gateway-metrics.submariner-operator.svc.cluster.local:8080/metrics"
+	gnCurlMetricsCommand = "curl -I -m 10 submariner-globalnet-metrics.submariner-operator.svc.cluster.local:8081/metrics"
+)
+
+var skipSingleNodeCheck bool
+
+func checkMetricsConfig(clusterInfo *cluster.Info, status reporter.Interface) bool {
+	result := checkComponentMetrics(clusterInfo, status, "gateway", gwCurlMetricsCommand)
+
+	if clusterInfo.Submariner.Spec.GlobalCIDR != "" {
+		result = checkComponentMetrics(clusterInfo, status, "globalnet", gnCurlMetricsCommand) && result
+	}
+
+	return result
+}
+
+func checkComponentMetrics(clusterInfo *cluster.Info, status reporter.Interface, component, command string) bool {
+	status.Start("Checking if %s metrics are accessible from non-gateway nodes", component)
+	defer status.End()
+
+	if !skipSingleNodeCheck {
```

Comment:
It's better to avoid global variables. If we want to cache the flag we should do it in `cluster.Info`.